src/modules/directory.rs

```diff
@@ -161,6 +152,57 @@ fn to_fish_style(pwd_dir_length: usize, dir_string: String, truncated_dir_string
         .join("/")
 }
 
+#[cfg(target_os = "windows")]
+fn to_separator(path: &Path, separator: &str) -> Option<String> {
```

Comment:
1. Have you investigated unifying the windows and -nix implementations? Gating platform behaviour in `#[cfg(windows)]` and/or `if cfg!(windows)`?
2. I think this will never return `None`, maybe switch to plain `String`?

Reply:
1: Not sure, windows&unix paths have very unique characteristics, like absolute paths always starting with / on Linux and on Windows they are relative to the current drive...
2: Sure, done.